unreleased_history/behavior_changes/wal_sync_on_close.md

```diff
@@ -0,0 +1 @@
+DB `Close()` will sync the WALs now
```

Comment:
It seems to me that we're trading performance for consistency here: a clean closing of the database now ensures we're safe from host crashes but closing itself can become much more expensive, right? This feels like a big change; I'm wondering if we should make it optional (and disabled by default to prevent surprises for clients).

Reply:
That's a good point - let me think a bit more about this. I also wondered how much close() performance matter since adding a DB options is "expensive" in usability.

Reply:
With more research, i realized this in our API. So it seems like the responsibility of WAL sync on users has been documented   @ajkr 

```
This will not fsync the WAL files.
  // If syncing is required, the caller must first call SyncWAL(), or Write()
  // using an empty write batch with WriteOptions.sync=true.
 ```

Reply:
Let's not block our crash test by this discussion - https://github.com/facebook/rocksdb/pull/12567

Reply:
> With more research, i realized this in our API. So it seems like the responsibility of WAL sync on users has been documented @ajkr

I think good documentation cannot make up for poor API quality. In my opinion, a safe default for close would be to sync the data that we will have no future opportunities to sync.

Reply:
I don't have a strong opinion on this but I think in general we don't explicitly sync the WAL files we're done with unless we absolutely have to. One thing this means is that if we're going to do it in this case, we'll probably have to call `SyncClosedLogs` to make sure any earlier unsynced WALs are also synced.

Reply:
Fixed with SyncWAL()